### ISSUE TITLE
Исследовать модели и подходы для семантической сегментации

### ISSUE THREAD

**Issue by EgorRazumilov** [Aorta-Detectors/Aorta-Detection#3]
Нужно сделать пайплайн для семантической сегментации аорты в организме.

Что хочется видеть по итогу:
**Готовый удобный для Вас пайплайн, для максимально, насколько это возможно, быстрого прототипирования и обучения модели** (насколько я понимаю, каких-то больших мощностей у Вас нет, нужно написать хороший Jupyter-Notebook в Google Colab)

Какие примерно степы нужно реализовать для достижения цели:

- [ ] Предобработка данных: работать надо будет с разными датасетами, они в разном формате, хочется достаточно быстро приспосабливать предобработку под каждый из них (здесь и загрузка данных, и написание наследников классa torch.data.utils.Dataset, и возможные аугментации к загруженным данным)
- [ ] Разобраться с базовыми нейронными архитектурами (по типу Unet) и реализовать их, для использования в датасетах (полагаю, что-то тяжелее мы в colab пока не засунем, поэтому пока работаем с базой)
- [ ] Написать хороший training cycle (возможности логгирования в колабе, очевидно, ограничены, но точно хочется видеть графики трейнового и валидационного лоссов, метрик сегментации, например, IoU, возможности early stopping, и learning rate scheduler)
- [ ] После проведения базовых экспериментов хочется видеть отчет о начальных результатах по поводу экспериментов: например, метрики на тестовых данных по каждому эксперименту, также хочется посмотреть, если останется время, результаты экспериментов по смешению датасетов: обучились на одном, потестировали на другом, записали метрики
(в боксах можно ставить галочки по завершении задачи :) )

По поводу данных, продублирую сюда:
1. SegTHOR (https://competitions.codalab.org/competitions/21145#learn_the_details-dataset). Ссылка на скачивание данных https://competitions.codalab.org/my/datasets/download/965273d5-ef82-4a57-b68f-bc530316fe89
2. Synapse. Надо зарегистрироваться, чтоб скачать https://www.synapse.org/#!Synapse:syn3193805/files/
3. TBAD dataset https://drive.google.com/drive/folders/1nTFusCpvn4LNwrzgs0mtefasf7d6tY81. Информация о spacing лежит в Excel файле
4. Огромный датасет из 1204 КТ исследований, размечены 104 анатомических структур (в том числе и аорта), но за качество неизвестно. https://zenodo.org/record/6802614#.ZAnOD3ZByUl
